### PR TITLE
feat: replace theme-ui with chakra, implement color modes

### PR DESCRIPTION
**Pull request by moripen**

### CODE REVIEW

File: Chakra.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  import { GetServerSidePropsContext } from "next";
+  import { ReactNode } from "react";
+  import theme from "./src/theme";
+  
+  interface ChakraProps {
+    cookies?: string;
+    children: ReactNode;
+  }
+  
+  export const Chakra = ({ children, cookies }: ChakraProps) => {
```
Comment (braaar) [inline Chakra.tsx:15]: Could you explain a bit what this component does? It's not in use, currently, as far as I can tell

**Review by braaar — CHANGES_REQUESTED** [review]

File: src/components/layout/layout.tsx [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
 import { SEO } from '../seo';
 import { Footer } from './footer/footer';
 import { Header } from './header';
 
-export const Layout: React.FC = ({ children }) => (
+type Props = {
+  children?: ReactNode;
+};
+
+export const Layout = ({ children }: Props) => (
```
[the remark this answers is not in the document]
Comment (moripen) [reply src/components/layout/layout.tsx:11]: Agree! This was an unfortunate byproduct of many hours of bugsquashing

File: package.json [hunk 3]
```diff
@@ -11,16 +11,20 @@
...
+    "@chakra-ui/react": "^2.4.3",
+    "@chakra-ui/theme-tools": "^2.0.14",
+    "@emotion/react": "^11.10.5",
+    "@emotion/styled": "^11.10.5",
     "@mdx-js/react": "^1.6.22",
     "@theme-ui/color": "^0.13.1",
     "@theme-ui/prism": "^0.13.1",
-    "next": "12.1.0",
+    "framer-motion": "^7.8.0",
+    "next": "13.0.6",
```
[the remark this answers is not in the document]
Comment (moripen) [reply package.json:24]: Agree, but yet again it was the result of the bugsquash patrol that got out of hand. Will try to separate it out - tomorrow(today)

File: .eslintrc.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
   "extends": ["@bjerk/eslint-config", "plugin:@next/next/recommended"],
   "overrides": [
     {
-      "files": "src/pages/**/*.tsx",
+      "files": "pages/**/*.tsx",
```
Comment (braaar) [inline .eslintrc.json:5]: Any particular reason why this change was made in this PR?

File: .vscode/settings.json [hunk 5]
```diff
@@ -0,0 +1,4 @@
+{
+  "search.useIgnoreFiles": false,
+  "search.exclude": {}
+}
```
Comment (braaar) [inline .vscode/settings.json:4]: Let's add this file to .gitignore

File: components/about/contact-person.tsx [hunk 6]
```diff
@@ -48,35 +54,47 @@ export const ContactPerson: React.FC<ContactPersonProps> = ({
...
+              base: 'auto',
+              md: '3rem 2rem 1.5rem 2rem 2rem 2rem',
             }}
           >
-            <Text sx={{ fontSize: 2, fontWeight: '600', mt: '20px' }}>
+            <Text
+              sx={{
+                fontSize: 'base',
+                fontWeight: 'semibold',
+                mt: '20px',
```
Comment (braaar) [inline components/about/contact-person.tsx:66]: Do we need more spacing options in the theme file so we can avoid using `px` here?

File: components/booking.tsx [hunk 7]
```diff
@@ -14,13 +14,19 @@ export const Booking: React.FC<BookingProps> = ({ label, variant }) => {
...
+        variant={variant}
         sx={{
           cursor: 'pointer',
+          color: 'dark100',
+          p: 5,
+          fontWeight: 'normal',
+          boxShadow:
+            '2px 4px 11px rgba(0, 0, 0, 0.1), inset 3px 4px 9px rgba(255, 255, 255, 0.15)',
+          borderRadius: '30px',
+          width: 'fit-content',
```
Comment (braaar) [inline components/booking.tsx:29]: Where is all this coming from? Shouldn't this stuff be in the theme file?

File: components/layout/layout.tsx [hunk 8]
```diff
@@ -1,10 +1,14 @@
...
+import { ReactNode } from 'react';
+import { Flex } from '@chakra-ui/react';
 import { SEO } from '../seo';
 import { Footer } from './footer/footer';
 import { Header } from './header';
 
-export const Layout: React.FC = ({ children }) => (
+type Props = {
+  children?: ReactNode;
+};
```
Comment (braaar) [inline components/layout/layout.tsx:9]:
```suggestion
```

File: components/layout/layout.tsx [hunk 9]
```diff
@@ -1,10 +1,14 @@
...
+
+export const Layout = ({ children }: Props) => (
```
Comment (braaar) [inline components/layout/layout.tsx:11]:
```suggestion
export const Layout : React.FC<React.PropsWithChildren> = ({ children }) => (
```

File: components/page-section.tsx [hunk 10]
```diff
@@ -1,7 +1,9 @@
-import { Box } from 'theme-ui';
+import { ReactNode } from 'react';
+import { Box } from '@chakra-ui/react';
 
 export interface PageSectionProps {
   backgroundColor?: 'green';
+  children?: ReactNode;
```
Comment (braaar) [inline components/page-section.tsx:6]:
```suggestion
```

File: components/page-section.tsx [hunk 11]
```diff
@@ -1,7 +1,9 @@
...
 }
 
 export const PageSection: React.FC<PageSectionProps> = ({
```
Comment (braaar) [inline components/page-section.tsx:9]:
```suggestion
export const PageSection: React.FC<React.PropsWithChildren<PageSectionProps>> = ({
```

File: package.json [hunk 12]
```diff
@@ -7,30 +7,35 @@
...
+    "@emotion/styled": "^11.10.5",
     "@mdx-js/react": "^1.6.22",
-    "@theme-ui/color": "^0.13.1",
-    "@theme-ui/prism": "^0.13.1",
-    "next": "12.1.0",
-    "next-theme-ui": "^1.2.0",
-    "react": "17.0.2",
+    "framer-motion": "^7.8.0",
+    "next": "13.0.6",
+    "react": "18.2.0",
```
Comment (braaar) [inline package.json:25]: Bumping react to 18 seems out of scope for this PR

File: package.json [hunk 13]
```diff
@@ -7,30 +7,35 @@
...
-    "react-dom": "17.0.2",
-    "react-intl": "^5.24.4",
-    "theme-ui": "^0.13.1"
+    "react-dom": "18.2.0",
+    "react-intl": "^5.24.4"
   },
   "prettier": "@cobraz/prettier",
   "devDependencies": {
-    "@bjerk/eslint-config": "^1.0.0",
+    "@bjerk/eslint-config": "^3.0.1",
```
Comment (braaar) [inline package.json:32]: This is out of scope for this PR. This (double) major version bump produces breaking changes!

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (braaar) [reply package.json:24]: Moving `pages` back into `src` will help with the readability of the diff on this PR, as well.

**Review by braaar — COMMENTED** [review]
All in all, the theming-related changes look solid, but some irrelevant changes have snuck in here and there.

File: .vscode/settings.json (hunk 5 above)
Comment (moripen) [reply .vscode/settings.json:4]: I altered the search parameters earlier to find some definitions in the node modules, but yes it belongs in gitignore - fixing it now

File: components/about/contact-person.tsx (hunk 6 above)
Comment (moripen) [reply components/about/contact-person.tsx:66]: I was going to do something similar to the thing with fontsize - adaptible spacing depending on page size using this: https://utopia.fyi/space/calculator/ tool. But i didnt get around to it yet :)

**Comment by moripen** [comment]
Deprecated: Done in PR#479
